pkg/volume/flexvolume/probe.go

```diff
@@ -38,12 +40,36 @@ func ProbeVolumePlugins(pluginDir string) []volume.VolumePlugin {
 		// then, executable will be pluginDir/dirname/cifs
 		if f.IsDir() {
 			execPath := path.Join(pluginDir, f.Name())
-			plugins = append(plugins, &flexVolumePlugin{
-				driverName:          utilstrings.UnescapePluginName(f.Name()),
-				execPath:            execPath,
-				runner:              exec.New(),
-				unsupportedCommands: []string{},
-			})
+
+			attachablePlugin := &flexVolumeAttachablePlugin{
+				flexVolumePlugin: flexVolumePlugin{
+					driverName:          utilstrings.UnescapePluginName(f.Name()),
+					execPath:            execPath,
+					runner:              exec.New(),
+					unsupportedCommands: []string{},
+				},
+			}
+
+			// Check whether the plugin is attachable.
+			ok, err := attachablePlugin.isAttachable()
+			if err != nil {
+				glog.Errorf("Unable to probe plugin: %s", attachablePlugin.driverName)
```

Comment:
I recommend including the isAttachable() error in the message to give a reason why probe is not successful.